sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -163,11 +192,49 @@ class CodegenContext {
    *                 the list of default imports available.
    *                 Also, generic type arguments are accepted but ignored.
    * @param variableName Name of the field.
-   * @param initCode The statement(s) to put into the init() method to initialize this field.
+   * @param initFunc Function includes statement(s) to put into the init() method to initialize
+   *                 this field. The argument is the name of the mutable state variable.
    *                 If left blank, the field will be default-initialized.
+   * @param forceInline whether the declaration and initialization code may be inlined rather than
+   *                    compacted. Please set `true` into forceInline, if you want to access the
+   *                    status fast (e.g. frequently accessed) or if you want to use the original
+   *                    variable name
+   * @param useFreshName If false and inline is true, the name is not changed
+   * @return the name of the mutable state variable, which is either the original name if the
+   *         variable is inlined to the outer class, or an array access if the variable is to be
+   *         stored in an array of variables of the same type.
+   *         There are two use cases. One is to use the original name for global variable instead
+   *         of fresh name. Second is to use the original initialization statement since it is
+   *         complex (e.g. allocate multi-dimensional array or object constructor has varibles).
+   *         Primitive type variables will be inlined into outer class when the total number of
```

Comment:
Some sentences here are not related to `return` doc. We should put it to proper place, maybe move them above the param doc.